### ISSUE TITLE
Ignore nvim-treesitter builtin queries to consider only custom ones for a language

### ISSUE THREAD

**Issue by gjeusel** [nvim-treesitter/nvim-treesitter#1441]
I am not satisfied with the queries provided by nvim-treesitter for a specific language.

Adding my custom queries folder named `queries` in my runtimepath (even before nvim-treesitter) does not seem to unload nvim-treesitter provided queries, even though it loads mine.
One solution would be to overwrite each and every query defined inside nvim-treesitter for this language, but it's rather cumbersome.

Would it be possible to add an option in the `setup` enabling to ignore queries for a specific language from nvim-treesitter ?
Or is there any alternative already existing ?

Right now I'm just modifying the queries directly inside nvim-treesitter plugin sources

**Comment by theHamsta** [comment]
It can be done with `:h set_query()`. You could use the a modified code of `get_query`

```lua
  local query_files = require'vim.treesitter.query'.get_query_files(lang, query_name)
  -- filter qurey files here with vim.tbl_filter (not contain nvim-treesitter)
  local query_string = read_query_files(query_files)

  if #query_string > 0 then
    return require'vim.treesitter.query'.parse_query(lang, query_string)
  end
```

It could be either documented in the README or we could provide a custom filter function. Ideally `set_query` should accept a function and not only a string to get the query text.

The easiest option for a personal config that can be done right now is
```lua
    use {
      "nvim-treesitter/nvim-treesitter",
      config = function()
        if require "nvim-treesitter.parsers".has_parser("python") then
          local folds_query =
            [[
  [
    (function_definition)
    (class_definition)

    (while_statement)
    (for_statement)
    (if_statement)
    (with_statement)
    (try_statement)

    (import_from_statement)
    (parameters)
    (argument_list)

    (parenthesized_expression)
    (generator_expression)
    (list_comprehension)
    (set_comprehension)
    (dictionary_comprehension)

    (tuple)
    (list)
    (set)
    (dictionary)

    (string)
  ] @fold
  ]]
          require("vim.treesitter.query").set_query("python", "folds", folds_query)
        end
      end
    }
```
this fixes the fold queries for Python for me.

**Comment by theHamsta** [comment]
I reopened this issue since we have to document this better or provide an option directly in setup

**Comment by theHamsta** [comment]
Out of interest: which language do you want to fix?

**Comment by stsewd** [comment]
We have https://github.com/nvim-treesitter/nvim-treesitter#adding-queries, but I think that doesn't work anymore.

**Comment by gjeusel** [comment]
Yes indeed, I believe an option in the setup, documented, to give a list of languages queries to ignore from nvim-treesitter would be optimal.

Same as you ! Python folding for one. But in general foldings (vue - typescript - ...) as nvim-treesitter defines too many of them for my taste.

Also a few other queries related to highlighting. For example, in python:
```queries
; Identifier naming conventions
((identifier) @type
 (#match? @type "^[A-Z]"))
```

Will also match constants like `APP`:
```python
import flask

APP = flask.Flask(__name__)
```

Also, in my mind, the `__name__` should be considered a @constant.builtin

In short, I feel like nvim-treesitter defining a lot of queries is good as it serves as examples and inspiration to write its owns. But in the end, users should be encouraged to write their owns, and be given the choice to unload the ones from nvim-treesitter.

That's my 50cts !

**Comment by theHamsta** [comment]
I guess this APP can be fixed like in C++ 
```
((identifier) @type
 (#match? @type "^[A-Z].*[a-z]"))
```

**Comment by haolian9** [comment]
there is a special case for `(#match? @type "^[A-Z].*[a-z]")` needs to consider.

```python
APP = "flask app instance"
URL = collections.namedtuple('URL', 'scheme host port')
```

@theHamsta

**Comment by haolian9** [comment]
i think the quick yet dirty fix is:

```
; address #1546
(class_definition
  name: (identifier) @type)

; revert GH-1450, address namedtuple use case
((identifier) @type
 (#match? @type "^[A-Z]"))

; address #1441 (dirty fix which introduced hardcode and only for people using flask)
((identifier) @constant
  (#any-of? @constant "APP"))
```

eventually, I think it is better to keep the APP as constant for now; as soon as https://github.com/neovim/neovim/pull/14932 has been merged, user can do what they want
